nixos/modules/services/monitoring/karma.nix

```diff
@@ -0,0 +1,179 @@
+      default = null;
+      description = mdDoc ''
+        Provides Karma with a raw config file instead of a nix-generated file.
+      '';
+      example = "/etc/karma/karma.conf";
+    };
+
+    environment = mkOption {
+      type = types.attrs;
+      default = {};
+      description = mdDoc ''
+        Additional environment variables to provide to karma.
+      '';
+      example = {
+        ALERTMANAGER_URI = "https://alertmanager.example.com";
+        ALERTMANAGER_NAME= "single";
+      };
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Whether to open ports in the firewall needed for karma to function.
+      '';
+    };
+
+    extraOptions = mkOption {
+      type = with types; listOf str;
+      default = [];
+      description = mdDoc ''
+        Extra command line options.
+      '';
+      example = [
+        "--alertmanager.timeout 10s"
+      ];
+    };
+
+    address = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = mdDoc ''
+        Hostname or IP to listen on.
+      '';
+      example = "[::]";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8080;
+      description = mdDoc ''
+        HTTP port to listen on.
+      '';
+      example = 8182;
+    };
+
+    readTimeout = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = mdDoc ''
+        HTTP server request read timeout.
+      '';
+      example = "15s";
+    };
+
+    writeTimeout = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = mdDoc ''
+        HTTP server response write timeout.
+      '';
+      example = "15s";
+    };
+
+    urlPrefix = mkOption {
+        type = types.str;
+        default = "/";
+        description = mdDoc ''
+          URL root for karma, you can use this to serve from a location other than `/`.
+
+          This option is mostly useful when using karma behind a reverse proxy with other services on the same IP but different URL root.
+        '';
+        example = "/dashboard";
+    };
+
+    tls = {
+      cert = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        description = mdDoc ''
+          Path to a TLS certificate, enables listening on HTTPS instead of HTTP.
+        '';
+        example = "/etc/karma/certs/tls_cert.pem";
+      };
+
+      key = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        description = mdDoc ''
+          Path to a TLS key, required when `tls.cert` is set.
+        '';
+        example = "/etc/karma/certs/tls_key.key";
+      };
+    };
+
+    config = mkOption {
+      type = types.attrs;
+      default = {};
+      description = mdDoc ''
+        Karma dashboard configuration as nix attributes.
+
+        Reference: https://github.com/prymitive/karma/blob/main/docs/CONFIGURATION.md
```

Comment:
Users will love you if you make this link clickable directly from documentation.

Reply:
Oh. I thought that it would render as a clickable link. I'll change that.